src/Symfony/Component/Scheduler/Command/DebugCommand.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Scheduler\RecurringMessage;
+use Symfony\Component\Scheduler\ScheduleProviderInterface;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+use function Symfony\Component\Clock\now;
+
+/**
+ * Command to list/debug schedules.
+ *
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+#[AsCommand(name: 'debug:scheduler', description: 'List schedules and their jobs')]
+final class DebugCommand extends Command
+{
+    public function __construct(private ServiceProviderInterface $schedules)
+    {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('schedule', InputArgument::OPTIONAL, sprintf('The schedule name (one of "%s")', implode('", "', array_keys($this->schedules->getProvidedServices()))))
```

Comment:
You can pass the values for shell completion as last argument of this method. 

```suggestion
        $scheduleNames = array_keys($this->schedules->getProvidedServices());
        $this
            ->addArgument('schedule', InputArgument::OPTIONAL, sprintf('The schedule name (one of "%s")', implode('", "', $scheduleNames)), null, $scheduleNames)
```

Maybe define the `private array $scheduleNames` property in the constructor to factorize with the `execute` method.

Reply:
Good call!